schema/interface.ipv4.yml

```diff
@@ -15,22 +15,13 @@ properties:
     - static
   subnet:
     description:
-      This option defines a list of CONNECTED routes (with VRF id) in CIDR notation.
-    type: array
-    items:
-      type: object
-      properties:
-        prefix:
-          description:
-            Defines a CONNECTED route's prefix (network).
-          type: string
-          format: uc-cidr4
-          examples:
-          - 192.168.1.0/24
-        vrf:
-          description:
-            VRF id.
-          type: number
+      This option defines the static IPv4 of the logical interface in CIDR notation.
+      auto/24 can be used, causing the configuration layer to automatically use
+      and address range from globals.ipv4-network.
+    type: string
```

Comment:
-1;
These changes are necessary for the extended OLS schema / code to work: original AP schema is very limited in terms of configuring ipv4;
* Changed interfaces[].ipv4.subnet    (array of prefix+vrf) // This is list of connected routes
* Changed interfaces[].ipv4.gateway   (array of prefix+gateway+vrf) // This is list of nexthop routes (several routes with the same prefix/metric treated as ECMP)

List of connected routes - basically list of ipv4 addresses, since you can have multiple per iface

Reply:
I can accept that, if it is the case.
However, the schema and the client should be updated in unison, and the examples should validate against the schema.
The changes from number to integer are worth considering as well, especially if your configuration actually wants an integer value.
Anyway, will leave it for now, I can cancel the PR, I will just fix my local copy so it works and then pull this when it is sorted and aligned.

Reply:
Hi Mike;
Thanks for the reply and the changes;

I understand that you haven't seen the HLD (sorry for the confusion), it's just that i wanted to show you a snippet of why it's done like this (motivation);
As for multiple adresses: with these changes (subnet->array) we can utilize multiple different addresses with subnets configured, that was basically the whole premise of the change;
It's been deliberately changed, and i know that differs from original AP, but that's OK for our OLS fork;

As for examples - yes, they're outdated, my bad. Haven't looked at them for a while;

Changes from number to integer - these are good, and appreciated as well, thanks. I think these originally originate from the AP code itself i think? And the OLS-NOS started at first as a copy-cat of proto.c and main files of wlan-ap, and i think we just kept it that way at that point, and maybe didn't bother at the time to consider number->integer change.

Reply:
Hi Oleksandr,
No problem re HLD, I appreciate the clarification, it will help me when I go through other parts of the code and schema. Far from a switch expert or even an engineer, I am computer scientist who get's to play with hardware ;-) so I usually need a few mistakes to get it right.
My plan is to revert the subnet changes, back to array, and re-submit the PR with just the number to integer changes.
I may at the same time update the examples on the ols-ucentral-client as well, which would bring them into alignment, I just want to be able to point people to them as an example. I also like to use it to validate things.
Expect this new PR probably sometime on Monday.